projects/plugins/social/tests/php/test-class-meta-tags.php

```diff
@@ -0,0 +1,180 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+/**
+ * Meta tags testing.
+ *
+ * @package automattic/jetpack-social-plugin
+ */
+
+use Brain\Monkey;
+use WorDBless\BaseTestCase;
+
+/**
+ * Meta tags testing.
+ */
+class Meta_Tags_Test extends BaseTestCase {
+	/**
+	 * Post ID of the testing post.
+	 *
+	 * @var int $post
+	 */
+	protected static $post;
+
+	/**
+	 * ID of the test attachment (1200 pixels wide).
+	 *
+	 * @var int $attachment_id
+	 */
+	protected static $attachment_id;
+
+	/**
+	 * ID of the test attachment (100 pixels wide).
+	 *
+	 * @var int $attachment_id_small
+	 */
+	protected static $attachment_id_small;
+
+	/**
+	 * Initialize tests
+	 *
+	 * @throws \phpmock\MockEnabledException The mock exception.
+	 *
+	 * @before
+	 */
+	public function set_up() {
+		$this->meta_tags           = new Automattic\Jetpack\Social\Meta_Tags();
+		self::$post                = wp_insert_post(
+			array(
+				'post_title'   => 'hello',
+				'post_content' => 'world',
+				'post_status'  => 'publish',
+			)
+		);
+		self::$attachment_id       = $this->create_upload_object( __DIR__ . '/images/jetpack-logo-1200w.png' );
+		self::$attachment_id_small = $this->create_upload_object( __DIR__ . '/images/jetpack-logo-100w.png' );
+	}
+
+	/**
+	 * Reverting the testing environment to its original state.
+	 *
+	 * @after
+	 */
+	public function tear_down() {
+		Monkey\tearDown();
+	}
+
+	/**
+	 * Create a upload
+	 *
+	 * @param string  $file File path.
+	 * @param integer $parent Parent post ID.
+	 * @return integer
+	 */
+	public function create_upload_object( $file, $parent = 0 ) {
+		$contents = file_get_contents( $file ); //phpcs:ignore
+		$upload   = wp_upload_bits( basename( $file ), null, $contents );
+
+		$type = '';
+		if ( ! empty( $upload['type'] ) ) {
+			$type = $upload['type'];
+		} else {
+			$mime = wp_check_filetype( $upload['file'], null );
+			if ( $mime ) {
+				$type = $mime['type'];
+			}
+		}
+
+		$attachment = array(
+			'post_title'     => basename( $upload['file'] ),
+			'post_content'   => '',
+			'post_type'      => 'attachment',
+			'post_parent'    => $parent,
+			'post_mime_type' => $type,
+			'guid'           => $upload['url'],
+		);
+
+		// Save the data.
+		$id = wp_insert_attachment( $attachment, $upload['file'], $parent );
+
+		wp_update_attachment_metadata( $id, wp_generate_attachment_metadata( $id, $upload['file'] ) );
```

Comment:
Rather than having the images and generating data around them in the tests, could we have the metadata as a fixture and insert in the setup? It might be a bit more efficient.

Reply:
Can you share an example? What you're seeing here is just [a copy of Core's implementation](https://core.trac.wordpress.org/browser/branches/4.5/tests/phpunit/includes/testcase.php#L766).

Reply:
Ah gotcha. That would probably work too, but I'm inclined to keep the actual images—it's how most tests related to attachments seem to do it and, like you said, it'll also safeguard against changes in the metadata structure.